packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -109,6 +115,18 @@ export async function generateAppYml(context: MigrationContext): Promise<void> {
   await context.fsWriteFile(path.join(SettingsFolderName, Constants.appYmlName), appYmlString);
 }
 
+export async function updateLaunchJson(context: MigrationContext): Promise<void> {
+  const launchJsonPath = path.join(context.projectPath, Constants.launchJsonPath);
+  if (await fs.pathExists(launchJsonPath)) {
+    await context.backup(Constants.launchJsonPath);
+    const launchJsonContent = await fs.readFile(launchJsonPath, "utf8");
+    const result = launchJsonContent
+      .replace(/\${teamsAppId}/g, "${dev:teamsAppId}") // TODO: set correct default env if user deletes dev, wait for other PR to get env list utility
+      .replace(/\${localTeamsAppId}/g, "${local:teamsAppId}");
```

Comment:
@XiaofuHuang @kuojianlu @swatDong I updated local part as well, feel free to leave your comment or change the logic in other PRs.

Reply:
Add one more replacement `${localTeamsAppInternalId}` -> `${local:teamsAppInternalId}`, which is for M365 apps.

Reply:
Thanks, updated